functests/utils/discovery/discovery.go

```diff
@@ -1,12 +1,58 @@
 package discovery
 
 import (
+	"context"
 	"os"
 	"strconv"
+
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+	performancev1alpha1 "github.com/openshift-kni/performance-addon-operators/pkg/apis/performance/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // Enabled indicates whether test discovery mode is enabled.
 func Enabled() bool {
 	discoveryMode, _ := strconv.ParseBool(os.Getenv("DISCOVERY_MODE"))
 	return discoveryMode
 }
+
+// GetDiscoveryPerformanceProfile returns an existing profile in the cluster with the most nodes using it.
```

Comment:
Why? I don't understand the benefit of using a profile used by most nodes? What is the upside? How does it help?

Reply:
@marcel-apf  we need to device a deterministic (although not super clever here) mechanism to get a profile to test against if we are in discovery mode (we will probably want to test the one with most nodes in CI)
this is ofcourse can get clever later on if we wish to test all profiles (Con - Time consuming)